command/agent/consul/service_client.go

```diff
@@ -913,66 +913,56 @@ func (c *ServiceClient) serviceRegs(ops *operations, service *structs.Service, w
 	ops.regServices = append(ops.regServices, serviceReg)
 
 	// Build the check registrations
-	checkIDs, err := c.checkRegs(ops, id, service, workload)
+	checkRegs, err := c.checkRegs(id, service, workload)
 	if err != nil {
 		return nil, err
 	}
-	for _, cid := range checkIDs {
-		sreg.checkIDs[cid] = struct{}{}
+	for _, registration := range checkRegs {
+		sreg.checkIDs[registration.ID] = struct{}{}
+		ops.regChecks = append(ops.regChecks, registration)
 	}
+
 	return sreg, nil
 }
 
-// checkRegs registers the checks for the given service and returns the
-// registered check ids.
-func (c *ServiceClient) checkRegs(ops *operations, serviceID string, service *structs.Service,
-	workload *WorkloadServices) ([]string, error) {
+// checkRegs creates check registrations for the given service
+func (c *ServiceClient) checkRegs(serviceID string, service *structs.Service,
+	workload *WorkloadServices) ([]*api.AgentCheckRegistration, error) {
 
-	// Fast path
-	numChecks := len(service.Checks)
-	if numChecks == 0 {
-		return nil, nil
-	}
-
-	checkIDs := make([]string, 0, numChecks)
+	registrations := make([]*api.AgentCheckRegistration, 0, len(service.Checks))
 	for _, check := range service.Checks {
-		checkID := MakeCheckID(serviceID, check)
-		checkIDs = append(checkIDs, checkID)
-		if check.Type == structs.ServiceCheckScript {
-			// Skip getAddress for script checks
-			checkReg, err := createCheckReg(serviceID, checkID, check, "", 0)
-			if err != nil {
-				return nil, fmt.Errorf("failed to add script check %q: %v", check.Name, err)
-			}
-			ops.regChecks = append(ops.regChecks, checkReg)
-			continue
-		}
+		ip := ""
```

Comment:
nit: If we're not explicitly setting defaults here, any reason not to just use `var ip string` and `var port int`?

Reply:
I was waffling on this; I personally prefer relying on the defaults as you showed, but I wasn't sure whether that would be preferred here or not. I'm happy to make the change!

Reply:
Yeah its totally a nitpicky thing so feel free to change or merge as is your call.

I agree a `uint` would probably be better suited here and other places where port numbers are represented but this is what we have today :slightly_frowning_face: